src/types.rs

```diff
@@ -637,22 +669,11 @@ impl From<&dojo_types::primitive::Primitive> for Primitive {
             dojo_types::primitive::Primitive::U16(v) => Primitive::U16(v.unwrap_or(0)),
             dojo_types::primitive::Primitive::U32(v) => Primitive::U32(v.unwrap_or(0)),
             dojo_types::primitive::Primitive::U64(v) => Primitive::U64(v.unwrap_or(0)),
-            dojo_types::primitive::Primitive::U128(v) => {
-                if let Some(v) = v {
-                    Primitive::U128(v.to_be_bytes())
-                } else {
-                    Primitive::U128([0; 16])
-                }
-            }
+            dojo_types::primitive::Primitive::U128(v) => Primitive::U128(*v.unwrap_or_default()),
             dojo_types::primitive::Primitive::U256(v) => {
-                if let Some(v) = v {
-                    Primitive::U256(v.to_words())
-                } else {
-                    #[cfg(not(target_pointer_width = "32"))]
-                    return Primitive::U256([0; 4]);
-                    #[cfg(target_pointer_width = "32")]
-                    return Primitive::U256([0; 8]);
-                }
+                // Assuming `v` is a byte array or similar; adjust as needed
+                let bytes = v.unwrap_or_else(|| [0u8; 32]);
+                Primitive::U256(U256::from_bytes(bytes))
```

Comment:
what about 32 bit architectures? and does unwrapping "v" actually return a byte array?